File: excellence/services/api/models/extraction_result.py
from typing import List, Literal

from .base import CamelModel
from .table_model import TableModel
from .validation_result import ValidationResult


ExtractionMethod = Literal[
    "ocr",
    "cheerio",
    "playwright",
    "relay",
    "paste",
    "text",
    "gemini",
]


class ExtractionResult(CamelModel):
    job_id: str
    table_model: TableModel
    validation_result: ValidationResult
    overall_confidence: float
    warnings: List[str]
    extraction_method: ExtractionMethod

File: excellence/packages/extraction-core/src/models/ExtractionResult.ts
import type { TableModel } from './TableModel';
import type { ValidationResult } from './ValidationResult';

export interface ExtractionResult {
  jobId: string;
  tableModel: TableModel;
  validationResult: ValidationResult;
  overallConfidence: number;
  warnings: string[];
  extractionMethod:
    | 'ocr'
    | 'cheerio'
    | 'playwright'
    | 'relay'
    | 'paste'
    | 'text'
    | 'gemini';
}
